index.md

```diff
@@ -38,122 +70,646 @@ Below are the __four scrum ceremonies__ that occur in each sprint.
+
+JDK consists of development tools and run time environment (JRE). If you want to develop and run a java program, you need a JDK. It is a kit or complete package to develop, compile and execute a Java program. It means that it contains JRE and JRE contains JVM.
+
+### Java Naming and Directory Interface (JNDI)
+
+JNDI is an API specified in Java technology that provides naming and directory functionality to applications written in the Java programming language. It is designed especially for the Java platform using Java's object model. Using JNDI, applications based on Java technology can store and retrieve named Java objects of any type. In addition, JNDI provides methods for performing standard directory operations, such as associating attributes with objects and searching for objects using their attributes.
+
+JNDI is also defined independent of any specific naming or directory service implementation. It enables applications to access different, possibly multiple, naming and directory services using a common API. Different naming and directory service providers can be plugged in seamlessly behind this common API. This enables Java technology-based applications to take advantage of information in a variety of existing naming and directory services, such as LDAP, NDS, DNS, and NIS(YP), as well as enabling the applications to coexist with legacy software and systems.
+
+Using JNDI as a tool, you can build new powerful and portable applications that not only take advantage of Java's object model but are also well-integrated with the environment in which they are deployed.
+
+### Java Runtime Environment (JRE)
+
+JRE is responsible for providing a runtime environment only. This is the reason it does not contain any development tools like compiler and debugger.
+JRE is platform dependent which makes Java a platform independent language.
+It consists of JVM and many software packages (Supported classes to run Java programs, Property files etc.) which help JVM to perform its tasks.
+
+### Java Virtual Machine (JVM)
+
+A JVM is a set of computer software programs and data structures which use a virtual machine model for the execution of other computer programs and scripts. The model used by a JVM accepts a form of computer intermediate language commonly referred to as Java bytecode.
+
+### JavaScript Object Notation (JSON)
+
+JavaScript Object Notation is a lightweight syntax containing objects and arrays, usually used (instead of XML) to return information from a REST API.
+It's an open standard format that allows data to be transferred across the web by using attribute-value pairs
+
+![JSON](images/JSON.png)
+
 ## K
 
 ## L
 
+### Load Balancing 
+
+Load balancing keeps your website up even when the traffic spikes, by distributing all the incoming traffic across the server pool. It helps high-traffic websites serve thousands of requests from clients in a fast and reliable manner while returning the right text and images even with a high volume.
```

Comment:
Add an image.. and different possible load balancing patterns like - Round Robin, etc..